content/blog/espressif-ide-lsp-support/index.md

```diff
@@ -0,0 +1,222 @@
+
+
+## clangd setup
+
+The LSP C/C++ Editor is configured to work with LSP and clangd by default. Here are some additional setup instructions that might be helpful if you encounter any challenges.
+
+
+<!-- omit in toc -->
+### clangd server setup
+
+The clangd path is configured in the preferences. The *Drivers* path and *--compile-commands-dir* path will be set based on the selected target (e.g., esp32, esp32c6) and the project you’re building.
+
+However, if there are any issues with the configuration, see the [clangd configuration][clangd-config] document.
+
+[clangd-config]: https://github.com/espressif/idf-eclipse-plugin/blob/master/docs_readme/clangd_cdt_support.md#clangd-configuration
+
+
+<!-- omit in toc -->
+### `.clangd` configuration
+
+For a new project, a `.clangd` configuration file is created by default with the contents provided below. For an existing project, create a `.clangd` file in its root folder yourself and add required parameters.
+
+```yaml
+CompileFlags:
+  CompilationDatabase: build
+  Remove: [-m*, -f*]
+```
+
+- `CompileFlags`: This key indicates the start of a dictionary (or map) containing compilation flag settings.
+- `CompilationDatabase`: This key specifies the location of the compilation database, which, in our case, is the `compile_commands.json` file generated by CMake. The value indicates that `compile_commands.json` is located in the `build` directory relative to the root of the project. For other paths, see [Custom build directory](#custom-build-directory).
+
+  A compilation database (`compile_commands.json`) is a file that contains an array of command objects, each representing a single compilation unit, providing details like the compiler executable, the compiler flags used, the source files, and the working directory. This database is used by clangd to determine how the source code is compiled.
+
+- `Remove`: This key lists the patterns, meaning that any compilation flags that start with `-m` or `-f` should be removed from the compilation commands. This is needed to remove some errors reported by clangd because Espressif-IDE uses the **GCC toolchain** by default to build projects. If you choose to use the experimental **esp-clang toolchain**, this key is not needed.
+
+  Here are the errors you may find in the file if you haven’t added the remove flags as mentioned above.
+
+  {{< gallery >}}
+    <img src="assets/1_BrhDa-MIopi70f-iebSKRQ.webp" />
+  {{< /gallery >}}
+
+For information about other `.clangd` configuration options, see [Configuration](https://clangd.llvm.org/config).
+
+
+<!-- omit in toc -->
+### Custom build directory
+
+If you use a custom build directory:
+
+- Set thhttps://clangd.llvm.org/confige `CompilationDatabase` key in the `.clangd` config file
```

Comment:
Typo with the url and text, it should be https://clangd.llvm.org/config

Reply:
Thanks for noticing this! Fixed.